### PR TITLE
Organize components & recipes

### PR DESCRIPTION
**Pull request by caseymhunt**
## Summary
Eventually we will want an organized presentation of stories sorted hierarchically, logically, and alphabetically. This change proposes an organization and sorting for components (and recipes). 

The sorting is as follows:
1. Hierarchical (Components, Recipes)
2. Logical (Forms, etc.)
3. Alphabetically

Note: This change uses a contrived (fictional) "recipe" of a `Log In Form` for the purposes of demonstration. This recipe is only to be considered placeholder. 

## Before & After
Before  | After
------------ | -------------
Storybook sidebar, before this change | Storybook sidebar, after this change

## Note on recipes
Though this is a contrived example and does not reflect actual design decisions, by viewing the "Docs" page of this recipe, it becomes evident how easily we can begin to create a robust collection of UI recipes using only a small number of components (already built). 

Placeholder recipe and the code necessary to build it

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -1,18 +1,24 @@
-import React from 'react';
-import { ChakraProvider } from '@chakra-ui/react';
-import theme from '../lib/theme';
+import React from "react";
+import { ChakraProvider } from "@chakra-ui/react";
+import theme from "../lib/theme";
```
Comment (nearbycoder) [inline .storybook/preview.js:3]: We probably have a discrepancy in our local prettier setup. Should add a `.prettierrc` and target for singlequotes.

File: lib/components/button/Button.stories.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Button } from './Button';
 import { Icon } from '../icons';
 
 export default {
-  title: 'Button',
+  title: 'Components/Button',
```
Comment (nearbycoder) [inline lib/components/button/Button.stories.js:6]: Nice TIL you can add folder structures

**Review by nearbycoder — COMMENTED** [review]
Overall looks good, just an issue with prettier setups for doublequotes vs. single quotes. adding a prettierrc will fix it for the project for us.

File: .storybook/preview.js (hunk 1 above)
Comment (caseymhunt) [reply .storybook/preview.js:3]: ooh yes! can we get a shared `.prettierrc`? is there an org-wide rc-repo or something?

**Comment by nearbycoder** [comment]
@caseymhunt I noticed that the Link in the Form Helper Text is using the hover state based on the form control. We may want to not include a link directly in that text helper

File: lib/components/button/Button.stories.js (hunk 2 above)
Comment (caseymhunt) [reply lib/components/button/Button.stories.js:6]: This + the custom sort in the storybook preview config is just _soo cooool!! _

**Comment by caseymhunt** [comment]
> @caseymhunt I noticed that the Link in the Form Helper Text is using the hover state based on the form control. We may want to not include a link directly in that text helper

Oooooh! What a good find! This is good info...

@jaxatto Do you foresee a case where we support links inside of helper text on forms as I've demonstrated here with the forgot password example? If so, we may need to change some stuff. If this is NOT a pattern we want then I can just remove the link and we move on.

File: .storybook/preview.js (hunk 1 above)
Comment (nearbycoder) [reply .storybook/preview.js:3]: this is our current setup within laddertruck

```
{
  "trailingComma": "all",
  "singleQuote": true
}

```

Comment (caseymhunt) [reply .storybook/preview.js:3]: adding `.prettierrc` to the repo with this.

**Review by nearbycoder — APPROVED** [review]
LGTM!

**Comment by jaxatto** [comment]
> > @caseymhunt I noticed that the Link in the Form Helper Text is using the hover state based on the form control. We may want to not include a link directly in that text helper
> 
> Oooooh! What a good find! This is good info...
> 
> @jaxatto Do you foresee a case where we support links inside of helper text on forms as I've demonstrated here with the forgot password example? If so, we may need to change some stuff. If this is NOT a pattern we want then I can just remove the link and we move on.

That! Is a really good question. My immediate inkling would be no, but it looks really good in your example and logically makes sense. I would say at this moment we don't have a need for it.

**Comment by jaxatto** [comment]
I really like grouping components under "Forms/", a la [Polaris](https://polaris.shopify.com/components/forms/autocomplete#navigation) for example. I think we should aim to do this with more components as we continue to add, like Card under "Layout/" and Buttons under "Actions/" etc. It can be a little tricky at first when you're looking at it and looking for something specific, but it makes sense in the long run. Also, there's always search.

My one qualm is the long list of examples in the navigation-- they're obviously helpful, but I loathe that it increases the nav so much. Maybe this is a me problem?

**Comment by caseymhunt** [comment]
> Let's keep iterating on the organization of components as we build more out. I like your proposal for folders like "Layout" and "Actions" - we should make that change once we have more than one for each category. 👍🏼

actually, forget that. I went ahead and did a first pass of reorganization based on your suggestions. how do you feel about this @jaxatto  ?

stories sorted into more folders per design recommendation

**Pull request merged**